pkg/apis/v1alpha1/dnspolicy_types.go

```diff
@@ -21,16 +21,15 @@ import (
 	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
-	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
 )
 
 // DNSPolicySpec defines the desired state of DNSPolicy
 type DNSPolicySpec struct {
 
 	// +kubebuilder:validation:Required
 	// +required
-	TargetRef gatewayapiv1alpha2.PolicyTargetReference `json:"targetRef"`
+	ParentRef gatewayapiv1.ParentReference `json:"parentRef"`
```

Comment:
Why moving to [`ParentReference`](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/shared_types.go#L35)? Is that because of `sectionName`?

My understanding is that this type is for specifying a parent of the resource, not a target of a policy. If it's for the `sectionName`, now there's [`PolicyTargetReferenceWithSectionName`](https://github.com/kubernetes-sigs/gateway-api/blob/68194976485b269e68214b5a40e5951815bbb80b/apis/v1alpha2/policy_types.go#L67) that can be used.

Reply:
Hi @guicassolato my reasoning for moving is that it appeared to me that parentRef was replacing targetRef as targetRef wasn't available in v1. But I think that I might have misunderstood that usage. I'm digging into it a little bit deeper now.